### ISSUE TITLE
Reviews

### ISSUE THREAD

**Issue by wwmmqq** [wwmmqq/SemEval-2018-Task-2-Multilingual-Emoji-Prediction#1]
============================================================================ 
SemEval 2018 Reviews for Submission #99
============================================================================ 

Title: Shi at SemEval-2018 Task 2: An Effective Attention-Based Recurrent Neural Network Model for Emoji Prediction with Characters Gated Words
Authors: Shiyun Chen, wang maoquan and Liang He
============================================================================
                            REVIEWER #1
============================================================================

---------------------------------------------------------------------------
Reviewer's Scores
---------------------------------------------------------------------------
                         Appropriateness: Appropriate (most submissions)
                           Clarity (1-5): 4
                         Soundness (1-5): 4
                     Replicability (1-5): 4
                  Overall Recommendation: Accept
               Best Paper Recommendation: No

Detailed Comments
---------------------------------------------------------------------------
This paper proposes to use a Bi-LSTM encoder with attention-based pooling to form a fixed vector representation of each sentence. The representation is then used to perform the 20-way classification using a soft-max. 

The use of both character and word-level information when constructing word representations was very appealing. It was well motivated from observations in the data and it produced strong empirical results.

Introduction:
- @uers -> @user
- I wouldn't call your four observations "rules", they aren't rules and are definitely observations from the data.
- I don't agree with perspective one, just because you can't match words identically, doesn't mean a traditional model like Naive Bayes-SVM wouldn't work. 

Model:
- What is the purpose of a learnable function g to combine the word and character level representations? Why not just concatenate the representations together?
- The attention layer needs to be explained more clearly. It is a type of attention pooling, though it is described as almost an attention layer you may seen in Neural Machine Translation.
- Any reason to include a regularization term?

Training:
- The learning rate is divided by 10 for the first three epochs? That seems very, very aggressive learning rate decay. Why was this done? 
- There is little description of the actual model used. How many layers in your LSTM? How many hidden units? 
- How many epochs was the model trained for?
---------------------------------------------------------------------------

============================================================================
                            REVIEWER #2
============================================================================

---------------------------------------------------------------------------
Reviewer's Scores
---------------------------------------------------------------------------
                         Appropriateness: Appropriate (most submissions)
                           Clarity (1-5): 4
                         Soundness (1-5): 5
                     Replicability (1-5): 4
                  Overall Recommendation: Accept
               Best Paper Recommendation: No

Detailed Comments
---------------------------------------------------------------------------
===== Summary ====

The paper presents an attention-based recurrent neural network model for emoji prediction for English tweets. The neural network model proposed in the paper takes words in a tweet as the input text. It then uses a combination of pre-trained word and character level embeddings to represent a word in a tweet and a bidirectional LSTM model with an attention layer to represent a tweet (sentence). A softmax layer is used as the output layer which will predict an emoji based on the input text. The approach presented is evaluated against a series of baseline models. The best performance is achieved when character and word features are combined with attention mechanism. The results reported in the paper positions the system presented at the 9th position out of 48 systems competed for English Language-based emoji prediction challenge.

==== Positives =====

The authors provide an analysis of the features available in the training dataset and justify the reasoning behind the selection of neural network models used. They try to evaluate the proposed model against a series of baseline and show that their system achieves the best performance. The writing of the paper can be further improved but the presentation is clear for anyone to understand. The authors try to relate the results with the features they identified in the training dataset so that the readers can examine why certain emoji were predicted with high accuracy and why the prediction of some emoji was challenging. 

===== Comments for Further Improvement ====

If you want to further experiment with higher dimension vectors for words (300), Wijeratne et al. share a Twitter embedding model learned over a corpus of 147 million tweets that contained at least one emoji. You can download it from here - http://emojinet.knoesis.org/dataset.php
 

===== Minor Mistakes ====

In Section 1 Introduction, you write "multi-category undoubtedly". This can be further improved by writing "multi-category classification have undoubtedly".
Section 2.1, first sentence, "the given" ==> "a given". 
Section 2.1, first sentence, "x, and" ==> remove the word "and".
Section 3.1, "Date is split into trial, training and test" ==> "Dataset is split into trial, training, and test sets".
---------------------------------------------------------------------------

============================================================================
                            REVIEWER #3
============================================================================

---------------------------------------------------------------------------
Reviewer's Scores
---------------------------------------------------------------------------
                         Appropriateness: Appropriate (most submissions)
                           Clarity (1-5): 5
                         Soundness (1-5): 5
                     Replicability (1-5): 5
                  Overall Recommendation: Accept
               Best Paper Recommendation: No

Detailed Comments
---------------------------------------------------------------------------
The paper is well written. Overall, the system is described with all the necessary details. 
Authors also perform interesting evaluations on their system to understand the best approach, and to show the importance of using char representation of words and attention over the words.

It would be very interesting for the future to visualize the the g_word and g_char weights values, to understand when one or the other representation are used.

Remove "Email:" at the beginning

For the quotation mark use `` ... '' instead of " ... "

You could cite some paper using the weights for the words and characters representation, maybe this one:
@InProceedings{C16-1030,
  author = "Rei, Marek and Crichton, Gamal and Pyysalo, Sampo",
  title = "Attending to Characters in Neural Sequence Labeling Models",
  booktitle = "Proceedings of COLING 2016, the 26th International Conference on Computational Linguistics: Technical Papers",
  year = "2016",
  publisher = "The COLING 2016 Organizing Committee",
  location = "Osaka, Japan"
}

The paper of Barbieri et al 2017, where the char lstm is used to predict emojis, is cited with the arxiv version, the correct citation is the following:
@InProceedings{barbieri2017, 
author = "Barbieri, Francesco and Ballesteros, Miguel and Saggion, Horacio", 
title = "Are Emojis Predictable?", 
booktitle = "Proceedings of the 15th Conference of the European Chapter of the Association for Computational Linguistics: Volume 2, Short Papers", 
year = "2017", 
publisher = "Association for Computational Linguistics", 
location = "Valencia, Spain" }

Also, the paper providing pretrained word embeddings is this one and not Barbieri 2017:
@inproceedings{barbieri2016cosmopolitan,
  title={How cosmopolitan are emojis?: Exploring emojis usage and meaning over different languages with distributional semantics},
  author={Barbieri, Francesco and Kruszewski, German and Ronzano, Francesco and Saggion, Horacio},
  booktitle={Proceedings of the 2016 ACM on Multimedia Conference},
  pages={531--535},
  year={2016},
  organization={ACM}
}
---------------------------------------------------------------------------

============================================================================
                            REVIEWER #4
============================================================================

---------------------------------------------------------------------------
Reviewer's Scores
---------------------------------------------------------------------------
                         Appropriateness: Appropriate (most submissions)
                           Clarity (1-5): 5
                         Soundness (1-5): 5
                     Replicability (1-5): 5
                  Overall Recommendation: Accept
               Best Paper Recommendation: No

Detailed Comments
---------------------------------------------------------------------------
The authors provide a very good description of their system, appropriate for a system description paper. Adequate references are provided that are key to the methods that were used. 

The only disadvantage of the paper is that the authors do not provide any previous literature summary, that justifies the methods that they used for the task. I would suggest that the authors perform a thorough literature review of the specific task and write a Related Work section, with references to well-known papers that are related to the task.
---------------------------------------------------------------------------

--
SemEval 2018 - https://www.softconf.com/naacl2018/SemEval-2018
